castero/queue.py

```diff
@@ -63,12 +63,24 @@ def play(self) -> None:
         """Plays the first player in the queue.
         """
         if self.first is not None:
-            self.first.episode.played = True
             self._display.modified_episodes.append(self.first.episode)
             self.first.play()
             self.first.set_volume(self.volume)
             self.first.set_rate(float(Config["default_playback_speed"]))
 
+            if self.first.time == 0:
+                self._goto_progress()
+
+    def _goto_progress(self):
+        """Seek forward to progress from start of episode
+
+        Args:
+        """
+        progress = self.first.episode.progress
+        if progress is not None and progress != 0:
+            MILLISECONDS_IN_SECONDS = 1000
```

Comment:
This could be moved to helper for easier reading as there are many instances of seconds <> milliseconds conversion